tests/foreman/ui/test_docker.py

```diff
@@ -90,57 +160,77 @@ def test_create_multiple_docker_repo_multiple_products(self):
 
         @Feature: Docker
 
-        @Status: Manual
-
         """
+        with Session(self.browser) as session:
+            for _ in range(randint(2, 3)):
+                pr = entities.Product(organization=self.organization).create()
+                for _ in range(randint(2, 3)):
+                    name = gen_string('utf8')
+                    _create_repository(
+                        session,
+                        org=self.organization.name,
+                        name=name,
+                        product=pr.name,
+                    )
+                    self.navigator.go_to_products()
+                    self.products.search(pr.name).click()
+                    self.assertIsNotNone(self.repository.search(name))
 
-    @stubbed()
     @run_only_on('sat')
     def test_sync_docker_repo(self):
         """@Test: Create and sync a Docker-type repository
 
-        @Assert: A repository is created with a Docker repository
-        and it is synchronized.
+        @Assert: A repository is created with a Docker repository and it is
+        synchronized.
 
         @Feature: Docker
 
-        @Status: Manual
-
         """
+        repo_name = gen_string('alpha')
+        product = entities.Product(organization=self.organization).create()
+        with Session(self.browser) as session:
+            _create_repository(
+                session,
+                org=self.organization.name,
+                name=repo_name,
+                product=product.name,
+            )
+            self.assertIsNotNone(self.repository.search(repo_name))
+            # Synchronize it
+            self.navigator.go_to_sync_status()
+            synced = self.sync.sync_custom_repos(product.name, [repo_name])
+            self.assertTrue(synced)
 
-    @stubbed()
     @run_only_on('sat')
-    @data(
-        gen_string('alpha', 15),
-        gen_string('alphanumeric', 15),
-        gen_string('numeric', 15),
-        gen_string('latin1', 15),
-        gen_string('utf8', 15),
-        gen_string('html', 15),
-    )
-    def test_update_docker_repo_name(self, new_name):
+    def test_update_docker_repo_name(self):
         """@Test: Create a Docker-type repository and update its name.
 
-        @Assert: A repository is created with a Docker image and that its
-        name can be updated.
+        @Assert: A repository is created with a Docker image and that its name
+        can be updated.
 
         @Feature: Docker
 
-        @Status: Manual
-
         """
+        with Session(self.browser) as session:
+            for new_name in valid_data_list():
+                with self.subTest(new_name):
+                    name = gen_string('alpha')
+                    product = entities.Product(
```

Comment:
I agree with @abalakh on his suggestion.  Whenever possible, we need to move the pre-requisite steps out of the subTest context loop to save time and to avoid repeating unnecessary steps.  This is one of the benefits of using subTest over ddt.

@oshtaier Product and repo creation if moved out of the loop will save more time than 1.5 seconds since this is UI.  If the test just focusses on repo new updates, I do not see a point in creating the product/repo for every iteration.